### PR TITLE
feat(tests-integration): ban invokes with nonce=0

### PR DESCRIPTION
**Pull request by giladchase**
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1065)

### CODE REVIEW

**Review by elintul — APPROVED** [review]
:lgtm:

> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1065#-:-O7s1pyaEdPEu2PzRxJM:bnfp4nl)* status: all files reviewed, 2 unresolved discussions (waiting on @ayeletstarkware and @giladchase)

___
*[`crates/mempool_test_utils/src/starknet_api_test_utils.rs` line 312 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/1065#-O7s1bVEAKz3WLOnu7Z0:-O7s1bVEAKz3WLOnu7Z1:b-9bff3r) ([raw file](https://github.com/starkware-libs/sequencer/blob/90c145e324df2cd80ef1e361632cae7a2c83ca6f/crates/mempool_test_utils/src/starknet_api_test_utils.rs#L312)):*
> ```XML
>         assert_ne!(
>             nonce,
>             Nonce(Felt::ZERO),
> ```

Better than `Nonce::default()` IYO? Indeed more explicit.



_Code quote:_
```XML
Nonce(Felt::ZERO)
```
___
*[`crates/mempool_test_utils/src/starknet_api_test_utils.rs` line 313 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/1065#-O7s1hmCC3GnstV1tKDF:-O7s1hmCC3GnstV1tKDG:undefined) ([raw file](https://github.com/starkware-libs/sequencer/blob/90c145e324df2cd80ef1e361632cae7a2c83ca6f/crates/mempool_test_utils/src/starknet_api_test_utils.rs#L313)):*
> ```XML
>             nonce,
>             Nonce(Felt::ZERO),
>             "Cannot invoke an undeployed account: the first transaction of every account must be
> ```





_Suggestion:_
```XML
invoke on behave of an
```

**Review by ayeletstarkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1065)* status: all files reviewed, 2 unresolved discussions (waiting on @elintul and @giladchase)

___
*[`crates/mempool_test_utils/src/starknet_api_test_utils.rs` line 312 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/1065#-O7s1bVEAKz3WLOnu7Z0:-O7wolue3txag9YVNhCY:b-3eypu4) ([raw file](https://github.com/starkware-libs/sequencer/blob/90c145e324df2cd80ef1e361632cae7a2c83ca6f/crates/mempool_test_utils/src/starknet_api_test_utils.rs#L312)):*
<details><summary><i>Previously, elintul (Elin) wrote…</i></summary><blockquote>

Better than `Nonce::default()` IYO? Indeed more explicit.
</blockquote></details>

can also use `nonce!(0)`

**Review by giladchase — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1065)* status: all files reviewed, 1 unresolved discussion (waiting on @ayeletstarkware and @elintul)

___
*[`crates/mempool_test_utils/src/starknet_api_test_utils.rs` line 312 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/1065#-O7s1bVEAKz3WLOnu7Z0:-O7xKGuI2d6kJWBpFj8Y:b-2qpvv9) ([raw file](https://github.com/starkware-libs/sequencer/blob/90c145e324df2cd80ef1e361632cae7a2c83ca6f/crates/mempool_test_utils/src/starknet_api_test_utils.rs#L312)):*
<details><summary><i>Previously, ayeletstarkware (Ayelet Zilber) wrote…</i></summary><blockquote>

can also use `nonce!(0)`
</blockquote></details>

I made it explicit because i want to emphasize that the nonce should be 0, rather than "just a default".
This is to adhere to the GW validation which checks this constraint.
In other words, I want to emphasize to the reader that it is 0, rather then default which is typically read as "don't care".

WDYT?

**Review by elintul — APPROVED** [review]
> Reviewed 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1065)* status: all files reviewed, 1 unresolved discussion (waiting on @ayeletstarkware)

**Review by giladchase — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1065)* status: 3 of 4 files reviewed, 1 unresolved discussion (waiting on @ayeletstarkware and @elintul)

___
*[`crates/mempool_test_utils/src/starknet_api_test_utils.rs` line 312 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/1065#-O7s1bVEAKz3WLOnu7Z0:-O7yQk-VDjRZMklFx_BK:b-sut9wa) ([raw file](https://github.com/starkware-libs/sequencer/blob/90c145e324df2cd80ef1e361632cae7a2c83ca6f/crates/mempool_test_utils/src/starknet_api_test_utils.rs#L312)):*
<details><summary><i>Previously, giladchase wrote…</i></summary><blockquote>

I made it explicit because i want to emphasize that the nonce should be 0, rather than "just a default".
This is to adhere to the GW validation which checks this constraint.
In other words, I want to emphasize to the reader that it is 0, rather then default which is typically read as "don't care".

WDYT?
</blockquote></details>

Discussed offline, switched to `nonce!(0)`.

**Review by elintul — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1065)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @giladchase)
